From fcfe69e94666fb8a21fcb0c474f7d350a5790cbf Mon Sep 17 00:00:00 2001
Message-Id: <fcfe69e94666fb8a21fcb0c474f7d350a5790cbf.1684999160.git.stefan@agner.ch>
In-Reply-To: <c2dc2a9940d1f9809403d4dfb28e1f942ef90bab.1684999160.git.stefan@agner.ch>
References: <c2dc2a9940d1f9809403d4dfb28e1f942ef90bab.1684999160.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 2 Mar 2023 03:12:46 +0100
Subject: [PATCH] [rest] explicitly set Connection header to close (#1774)

By default, HTTP 1.1 connections should stay open after a transaction.
However, that is not how the current REST server implementation behaves:
After each transaction the HTTP connection is being closed by the
server.

Set the Connection header to "close" to tell the client about this
behavior.

(cherry picked from commit 0e8af359120ee051314c1afb2bec2107c5bdf38d)
---
 src/rest/response.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/rest/response.cpp b/src/rest/response.cpp
index bcf63a6c31..2fd536c546 100644
--- a/src/rest/response.cpp
+++ b/src/rest/response.cpp
@@ -36,6 +36,7 @@
     "Access-Control-Allow-Headers, Origin,Accept, X-Requested-With, Content-Type, Access-Control-Request-Method, " \
     "Access-Control-Request-Headers"
 #define OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_METHOD "GET"
+#define OT_REST_RESPONSE_CONNECTION "close"
 
 namespace otbr {
 namespace rest {
@@ -52,6 +53,7 @@ Response::Response(void)
     mHeaders["Access-Control-Allow-Origin"]  = OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_ORIGIN;
     mHeaders["Access-Control-Allow-Methods"] = OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_METHOD;
     mHeaders["Access-Control-Allow-Headers"] = OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_HEADERS;
+    mHeaders["Connection"]                   = OT_REST_RESPONSE_CONNECTION;
 }
 
 void Response::SetComplete()
-- 
2.40.1

